torch/testing/_internal/common_methods_invocations.py

```diff
@@ -3021,6 +3021,31 @@ def generate_samples():
     return list(generate_samples())
 
 
+def sample_inputs_lu_solve(op_info, device, dtype, requires_grad=False, **kwargs):
+    from torch.testing._internal.common_utils import random_fullrank_matrix_distinct_singular_value
+
+    batches = [(), (0, ), (2, )]
+    ns = [5, 3, 0]
+    nrhs = [0, 1, 6]
+
+    def generate_samples():
+        for n, batch, rhs in product(ns, batches, nrhs):
+            a = random_fullrank_matrix_distinct_singular_value(n, *batch, dtype=dtype, device=device)
+            lu, pivs = a.lu()
+            lu.requires_grad = requires_grad
+            b = torch.randn(*batch, n, rhs, dtype=dtype, device=device)
+            b.requires_grad = requires_grad
+            yield SampleInput(b, args=(lu, pivs))
```

Comment:
Nit here is that each sample input should be independent of others, so for these you'll want to regenerate the tensors

What's going on with the `requires_grad` setting here? This would probably benefit from a comment

Reply:
Fixed!